tests/unit-tests/test-class-teacher.php

```diff
@@ -385,4 +385,57 @@ public function testAddCoursesToAuthorArchive() {
 
 	}
 
+	public function testGetTeachersAndAuthors() {
+		$this->create_use_with_role( 'subscriberName', 'subscriberPass', 'subscriber@email.com', 'subscriber' );
+		$this->create_use_with_role( 'subscriberBName', 'subscriberBPass', 'subscriberB@email.com', 'subscriber', '_b' );
+		$this->create_use_with_role( 'teacher', 'teacherPass', 'teacher@email.com', 'teacher' );
+		$this->create_use_with_role( 'teacherB', 'teacherBPass', 'teacherB@email.com', 'teacher', '_b' );
+		$this->create_use_with_role( 'teacherC', 'teacherCPass', 'teacherC@email.com', 'teacher', '_c' );
+		$this->create_use_with_role( 'editor', 'editorPass', 'editor@email.com', 'editor' );
+		$this->create_use_with_role( 'administrator', 'administratorPass', 'administrator@email.com', 'administrator' );
+		$users_with_edit_courses_rights_ids = Sensei()->teacher->get_teachers_and_authors();
+		$users                              = get_users(
+			[
+				'blog_id' => $GLOBALS['blog_id'],
+				'include' => $users_with_edit_courses_rights_ids,
+			]
+		);
+		$roles                              = array_merge( ...array_column( $users, 'roles' ) );
+		$teachers                           = array_filter(
+			$roles,
+			function( $item ) {
+				return 'teacher' === $item;
+			}
+		);
+		$this->assertCount( 3, $teachers );
+		$administrators = array_filter(
+			$roles,
+			function( $item ) {
+				return 'administrator' === $item;
+			}
+		);
+		$this->assertCount( 2, $administrators );
+		$editors = array_filter(
+			$roles,
+			function( $item ) {
+				return 'editor' === $item;
+			}
+		);
+		$this->assertCount( 1, $editors );
+		$subscribers = array_filter(
+			$roles,
+			function( $item ) {
+				return 'subscriber' === $item;
+			}
+		);
+		$this->assertCount( 0, $subscribers );
```

Comment:
You can simplify this part by looping through the already created users. For example:

```php
foreach ( $subscribers as $user_id ) {
	$this->assertNotContains( $user_id, $users_with_edit_courses_rights_ids, 'Should not include users that don\'t have the `edit_courses` capability.' );
}

foreach ( array_merge( $admins, $editors, $teachers ) as $user_id ) {
	$this->assertContains( $user_id, $users_with_edit_courses_rights_ids, 'Should include users which have the `edit_courses` capability.' );
}
```

Also, you can consider adding assertion messages (third parameter). That helps explain what the test does and makes it easier to identify which test failed.

Reply:
yeah, all of this makes totally sense